nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-core/src/main/java/org/apache/nifi/provenance/ComponentIdentifierLookup.java

```diff
@@ -46,14 +48,15 @@ public ComponentIdentifierLookup(final FlowController flowController) {
 
     @Override
     public List<String> getComponentTypes() {
-        final Set<Class> procClasses = flowController.getExtensionManager().getExtensions(Processor.class);
+        final Set<ExtensionDefinition> procDefinitions = flowController.getExtensionManager().getExtensions(Processor.class);
 
-        final List<String> componentTypes = new ArrayList<>(procClasses.size() + 2);
+        final List<String> componentTypes = new ArrayList<>(procDefinitions.size() + 2);
         componentTypes.add(ProvenanceEventRecord.REMOTE_INPUT_PORT_TYPE);
         componentTypes.add(ProvenanceEventRecord.REMOTE_OUTPUT_PORT_TYPE);
 
-        procClasses.stream()
-            .map(Class::getSimpleName)
+        procDefinitions.stream()
+            .map(ExtensionDefinition::getImplementationClassName)
+            .map(className -> className.contains(".") ? StringUtils.substringAfterLast(className, ".") : className)
```

Comment:
The `StandardStatelessEngine` uses the same approach attempting to derive the simple class name, does it make sense to add a method on `ExtensionDefinition` to return the simple class name?

Reply:
To do that, we would have to either implement the logic of substringAfterLast in ExtensionDefinition (which isn't difficult but is probably uglier than repeating this trivial tertiary check twice) or bring in a dependency on commons-utils, and that dependency cannot exist there due to classloader hierarchy. So I think it makes more sense to just leave as-is for now.

Reply:
That makes sense, given it is only in two places, not a real concern right now.